libimage/copier.go

```diff
@@ -167,39 +167,45 @@ var storageAllowedPolicyScopes = signature.PolicyTransportScopes{
 	},
 }
 
-// getDockerAuthConfig extracts a docker auth config from the CopyOptions.  Returns
-// nil if no credentials are set.
-func (options *CopyOptions) getDockerAuthConfig() (*types.DockerAuthConfig, error) {
-	authConf := &types.DockerAuthConfig{IdentityToken: options.IdentityToken}
-
-	if options.Username != "" {
-		if options.Credentials != "" {
-			return nil, errors.New("username/password cannot be used with credentials")
-		}
-		authConf.Username = options.Username
-		authConf.Password = options.Password
-		return authConf, nil
-	}
+// getDockerAuthConfig extracts a docker auth config. Returns nil if
+// no credentials are set.
+func getDockerAuthConfig(name, passwd, creds, idToken string) (*types.DockerAuthConfig, error) {
+	numCredsSources := 0
 
-	if options.Credentials != "" {
-		split := strings.SplitN(options.Credentials, ":", 2)
+	if name != "" {
+		numCredsSources++
+	}
+	if creds != "" {
+		split := strings.SplitN(creds, ":", 2)
 		switch len(split) {
 		case 1:
-			authConf.Username = split[0]
+			name = split[0]
```

Comment:
`passwd = ""` here, to ensure that users can’t set (`creds = "user", name = "", password = "..."`)?

Possibly just
```go
name, passwd, _ = strings.Cut(creds, ":")
```
would nowadays be simpler than the existing code, and have the desired effect.